File: test-suite-docs-java/src/test/java/io/micronaut/testresources/docs/junitplatform/ScopeNamingStrategyTest.java
package io.micronaut.testresources.docs.junitplatform;

// tag::imports[]
import io.micronaut.test.extensions.junit5.ScopeHolder;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.micronaut.test.extensions.junit5.annotation.ScopeNamingStrategy;
import io.micronaut.test.extensions.junit5.annotation.TestResourcesScope;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
// end::imports[]

// tag::clazz[]
@MicronautTest
@TestResourcesScope(namingStrategy = ScopeNamingStrategy.TestClassName.class) // <1>
class ScopeNamingStrategyTest {

    @Test
    void testSomething() {
        assertEquals(ScopeNamingStrategyTest.class.getName(), ScopeHolder.get().orElse(null)); // <2>
    }
}
// end::clazz[]

File: test-suite-docs-kotlin/src/test/kotlin/io/micronaut/testresources/docs/junitplatform/ScopeNamingStrategyTest.kt
package io.micronaut.testresources.docs.junitplatform

// tag::imports[]
import io.micronaut.test.extensions.junit5.ScopeHolder
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import io.micronaut.test.extensions.junit5.annotation.ScopeNamingStrategy
import io.micronaut.test.extensions.junit5.annotation.TestResourcesScope
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
// end::imports[]

// tag::clazz[]
@MicronautTest
@TestResourcesScope(namingStrategy = ScopeNamingStrategy.TestClassName::class) // <1>
class ScopeNamingStrategyTest {

    @Test
    fun testSomething() {
        assertEquals(ScopeNamingStrategyTest::class.java.name, ScopeHolder.get().orElse(null)) // <2>
    }
}
// end::clazz[]
